kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManager.java

```diff
@@ -211,35 +213,73 @@ private Producer registerInPersistentTopic(PersistentTopic persistentTopic) {
         }
         CompletableFuture<Optional<PersistentTopic>> topicCompletableFuture = new CompletableFuture<>();
         brokerService.getTopicIfExists(topicName).whenComplete((t2, throwable) -> {
+            TopicName topicNameObject = TopicName.get(topicName);
             if (throwable != null) {
-                // The ServiceUnitNotReadyException is retriable so we should print a warning log instead of error log
-                if (throwable instanceof BrokerServiceException.ServiceUnitNotReadyException) {
-                    log.warn("[{}] Failed to getTopic {}: {}",
-                            requestHandler.ctx.channel(), topicName, throwable.getMessage());
-                    topicCompletableFuture.complete(Optional.empty());
+                if (topicNameObject.getPartitionIndex() == 0) {
+                    log.warn("Get partition-0 error [{}].", throwable.getMessage());
```

Comment:
Why not call `removeTopicManagerCache` in this case?

Reply:
It should be call removeTopicManagerCache in this case, fixed.